lib/chef/resource/locale.rb

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright:: 2011-2016, Heavy Water Software Inc.
+# Copyright:: 2016-2018, Chef Software Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific slanguage governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+
+class Chef
+  class Resource
+    class Locale < Chef::Resource
+      preview_resource true
+      resource_name :locale
+
+      description "Use the locale resource to set the system's locale"
+      introduced "14.5"
+
+      property :lang, String,
+               default: "en_US.utf8",
+               description: "set the default system language"
```

Comment:
1 more thing. We use these to auto generate the documentation so can you make sure all the description fields have sentences that are capitalized and end with periods.